### PR TITLE
Remove debug signal workaround and consume tinylicious broadcast-signal endpoint instead

### PR DESCRIPTION
**Pull request by ghoshkaj**
[AB#4722](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/4722)

The purpose of this PR is to mainline the dev [broadcast-signal branch.](https://github.com/microsoft/FluidFramework/tree/dev/external-data-prototyping)

This PR consumes the changes in the server code and showcase how to broadcast-signal mechnaism in the example external-data app. This PR is blocked on both the [server-lambdas changes](https://github.com/microsoft/FluidFramework/pull/16451) as well as the [tinylicious changes](https://github.com/microsoft/FluidFramework/pull/16452).

### CODE REVIEW

**Review by seanimam — DISMISSED** [review]
Everything looks good to me but I'd like to preface that I am missing the full context beyond this PR so try to get another approval prior to merging.

File: examples/external-data/src/mock-customer-service/service.ts [hunk 1]
```diff
@@ -16,18 +16,29 @@ import { assertValidTaskData, ITaskData } from "../model-interface";
...
 	externalTaskListId: string,
 ): void {
 	console.log(
 		`CUSTOMER SERVICE: External data has been updated. Notifying Fluid Service at ${fluidServiceUrl}`,
 	);
 
-	// TODO: we will need to add details (like ContainerId) to the message body or the url,
-	// so this message body format will evolve
-	const messageBody = JSON.stringify({ taskData, containerUrl, externalTaskListId });
+	// Stringify signal content one time more than POST body requirement
```
Comment (Josmithr) [inline examples/external-data/src/mock-customer-service/service.ts:29]: Thanks for the explanation! This is great!

File: examples/external-data/src/mock-customer-service/service.ts [hunk 2]
```diff
@@ -170,29 +183,34 @@ export async function initializeCustomerService(props: ServiceProps): Promise<Se
...
 	 *		externalTaskListId: string
 	 *	}
 	 * ```
-	 *
-	 * Note: Implementers choice --can choose to break up containerUrl into multiple pieces
-	 * containing tenantId, documentId and socketStreamURL separately and send them as a json
-	 * object. The URL also contains all this information so for simplicity I use the url here.
 	 */
 	expressApp.post("/register-session-url", (request, result) => {
 		// eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
```
Comment (Josmithr) [inline examples/external-data/src/mock-customer-service/service.ts:193]: Nit: Might be worth doing a single disable block for all of the lines doing request.body?.foo accesses.
E.g.
```typescript
		/* eslint-disable @typescript-eslint/no-unsafe-member-access */
		const documentId = request.body?.documentId as string;
		const tenantId = request.body?.tenantId as string;
		const externalTaskListId = request.body?.externalTaskListId as string;
		/* eslint-enable @typescript-eslint/no-unsafe-member-access */
```

File: examples/external-data/src/model-interface/index.ts [hunk 3]
```diff
@@ -192,7 +194,8 @@ export interface IBaseDocumentEvents extends IEvent {
  */
 export interface IBaseDocumentInitialState {
 	externalTaskListId: string;
-	containerUrl: IResolvedUrl;
+	tenantId: string;
```
Comment (Josmithr) [inline examples/external-data/src/model-interface/index.ts:197]: Nit: probably worth documenting these since they're part of the example

**Review by Josmithr — APPROVED** [review]
Looks good to me!

**Pull request closed without merging**